packages/astro/src/assets/utils/node/emitAsset.ts

```diff
@@ -42,6 +43,11 @@ export async function emitESMImage(
 		value: id,
 	});
 
+	// Attach file data for SVGs
+	if (fileMetadata.format === 'svg') {
+		emittedImage.contents = fileData;
+	}
```

Comment:
Would love @Princesseuh's take on this. Any expected footguns with including the file content for SVGs in the image metadata?

Reply:
If you have a lot of SVGs, this can be problematic for SSR since it'll get in your bundle, that's my only concerns

Reply:
Right. Unfortunately that will be unavoidable for this feature.

I think the DX improvement is worth the tradeoff, though!

Reply:
This is definitely outside my realm of expertise. I'll defer to whatever the consensus is here :)